app/uk/gov/hmrc/tai/controllers/income/IncomeController.scala

```diff
@@ -48,17 +53,72 @@ class IncomeController @Inject()(incomeService: IncomeService,
       } recoverWith taxAccountErrorHandler
   }
 
-  def taxCodeIncomesForYear(nino: Nino, year:TaxYear): Action[AnyContent] = authentication.async {
+  def taxCodeIncomesForYear(nino: Nino, year: TaxYear): Action[AnyContent] = authentication.async {
     implicit request =>
       incomeService.taxCodeIncomes(nino, year).map {
         case Seq() => NotFound
         case taxCodeIncomes => Ok(Json.toJson(ApiResponse(Json.toJson(taxCodeIncomes)(Writes.seq(taxCodeIncomeSourceWrites)), Nil)))
-      }recoverWith taxAccountErrorHandler
+      } recoverWith taxAccountErrorHandler
+  }
+
+  def matchedTaxCodeIncomesForYear(nino: Nino, year: TaxYear, incomeType: String, status: String): Action[AnyContent] = authentication.async {
+    implicit request =>
+
+      def filterMatchingEmploymentsToIncomeSource(employments: Seq[Employment], filteredTaxCodeIncomes: Seq[TaxCodeIncome]): Seq[IncomeSource] =
+        filteredTaxCodeIncomes.flatMap { income =>
+          employments.flatMap(emp =>
+            income.employmentId.fold(Seq.empty[IncomeSource]) {
+              id => if (id == emp.sequenceNumber) Seq(IncomeSource(income, emp)) else Seq.empty[IncomeSource]
+            }
+          )
+        }
+
+      def filterTaxCodeIncomes(taxCodeIncomes: Seq[TaxCodeIncome], incomeStatus: String): Seq[TaxCodeIncome] = {
+        if (incomeStatus == Live.toString) {
+          taxCodeIncomes.filter(income => income.status == Live && income.componentType.toString == incomeType)
+        }
+        else {
+          taxCodeIncomes.filter(income => income.status != Live && income.componentType.toString == incomeType)
+        }
+      }
+
```

Comment:
Is this the same as           

`taxCodeIncomes.filter(income => income.status == incomeStatus && income.componentType.toString == incomeType)`